Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -247,6 +245,14 @@ public class DocumentIterator: IteratorProtocol {
         return values
     }
 
+    /// Returns an array of KeyValue pairs from the iterator's current position to the end. The iter will be exhausted
+    /// after this property is accessed.
```

Comment:
My understanding of this is: 
No, because non-destructive iteration just means that iterating it once does not change the Sequence/Collection itself or its contents, nor does it change the results of future iterations.

The only thing having an exhaustible iterator violates is infinite sequence, which a `Document` is not.